blueprints/data-solutions/data-platform-minimal/variables.tf

```diff
@@ -0,0 +1,225 @@
+          memory_gb  = number
+          storage_gb = number
+          count      = number
+        }
+        ), {
+        cpu        = 0.5
+        memory_gb  = 1.875
+        storage_gb = 1
+        count      = 1
+      })
+      web_server = optional(object(
+        {
+          cpu        = number
+          memory_gb  = number
+          storage_gb = number
+        }
+        ), {
+        cpu        = 0.5
+        memory_gb  = 1.875
+        storage_gb = 1
+      })
+      worker = optional(object(
+        {
+          cpu        = number
+          memory_gb  = number
+          storage_gb = number
+          min_count  = number
+          max_count  = number
+        }
+        ), {
+        cpu        = 0.5
+        memory_gb  = 1.875
+        storage_gb = 1
+        min_count  = 1
+        max_count  = 3
+      })
+    }))
+  })
+  default = {
+    environment_size = "ENVIRONMENT_SIZE_SMALL"
+    software_config = {
+      image_version = "composer-2-airflow-2"
+    }
+    workloads_config = {
+      scheduler = {
+        cpu        = 0.5
+        memory_gb  = 1.875
+        storage_gb = 1
+        count      = 1
+      }
+      web_server = {
+        cpu        = 0.5
+        memory_gb  = 1.875
+        storage_gb = 1
+      }
+      worker = {
+        cpu        = 0.5
+        memory_gb  = 1.875
+        storage_gb = 1
+        min_count  = 1
+        max_count  = 3
+      }
+    }
+  }
+}
+
+variable "data_catalog_tags" {
+  description = "List of Data Catalog Policy tags to be created with optional IAM binging configuration in {tag => {ROLE => [MEMBERS]}} format."
+  type        = map(map(list(string)))
+  nullable    = false
+  default = {
+    "3_Confidential" = null
+    "2_Private"      = null
+    "1_Sensitive"    = null
+  }
+}
+
+variable "data_force_destroy" {
+  description = "Flag to set 'force_destroy' on data services like BiguQery or Cloud Storage."
+  type        = bool
+  default     = false
+}
+
+variable "groups" {
+  description = "User groups."
+  type        = map(string)
+  default = {
+    data-analysts  = "gcp-data-analysts"
+    data-engineers = "gcp-data-engineers"
+    data-security  = "gcp-data-security"
+  }
+}
+
+variable "location" {
+  description = "Location used for multi-regional resources."
+  type        = string
+  default     = "eu"
+}
+
+variable "network_config" {
+  description = "Shared VPC network configurations to use. If null networks will be created in projects with preconfigured values."
+  type = object({
+    host_project      = string
+    network_self_link = string
+    subnet_self_links = object({
+      processing_dataproc = string
+      processing_composer = string
+    })
+    composer_ip_ranges = object({
+      cloudsql   = string
+      gke_master = string
+    })
+    composer_secondary_ranges = object({
+      pods     = string
+      services = string
+    })
+    # web_server_network_access_control = list(string)
+  })
+  default = null
+}
```

Comment:
- Why split the secondary ranges in another object?
- Make non-nullable and you avoid `try`
- Might be a good time to allow [PSC for the Cloud SQL instance](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/composer_environment#cloud_composer_connection_subnetwork).

Consider using something like this:
```hcl
variable "network_config" {
  description = "Shared VPC network configurations to use. If null networks will be created in projects with preconfigured values."
  type = object({
    host_project      = string
    network_self_link = string
    subnet_self_links = object({
      processing_dataproc = string
      processing_composer = string
    })
    composer_ip_ranges = object({
      connection_subnetwork = optional(string)
      cloud_sql             = optional(string)
      gke_master            = string
      pods_range_name       = string
      services_range_name   = string
    })

    # web_server_network_access_control = list(string) --> moved to composer_config.web_server_config
  })
  nullable = false
  validation {
    condition     = (var.network_config.composer_ip_ranges.cloud_sql == null) != (var.network_config.composer_ip_ranges.connection_subnetwork == null)
    error_message = "One, and only one, of `network_config.composer_ip_ranges.cloud_sql` or `network_config.composer_ip_ranges.connection_subnetwork` must be specified."
  }
}
```

Reply:
Good point, done